staging/src/k8s.io/component-base/metrics/metric.go

```diff
@@ -234,22 +233,3 @@ func (noopMetric) Desc() *prometheus.Desc            { return nil }
 func (noopMetric) Write(*dto.Metric) error           { return nil }
 func (noopMetric) Describe(chan<- *prometheus.Desc)  {}
 func (noopMetric) Collect(chan<- prometheus.Metric)  {}
-
-type noopObserverVector struct{}
```

Comment:
cc @logicalhan

This seems fine to me if it is unused

Reply:
It is used, why are we removing this. This is going to break stuff.

Reply:
how is it used? removing passed type checks and I don't see any other references in source

Reply:
Oh nvm, it doesn't seem to be used.

Reply:
Actually I think we need this, there seems to be a bug in an implemented wrapper.

Reply:
It's supposed to use this.

Reply:
I suppose we can delete it for now and resurrect this when we actually fix the bug though.

Reply:
since we actually want to backport the lint changes associated with go1.20, let's not churn removal/readd... we can wait a day to merge this to let the fix go in first

Reply:
The fix is invasive, I would just remove this for now and fix forwards.

Reply:
ok